### ISSUE TITLE
Hidden variable/definition when same named enum defined

### ISSUE THREAD

**Issue by oconnor0** [aardappel/lobster#278]
Running this lobster program with https://github.com/aardappel/lobster/releases/tag/v2023.13 produces no warnings:

```
enum foo:
    bar

let bar = 1
print "bar {bar}"
```

And this output:

```
bar bar
```

I've noticed shadowing and redefinition warnings/errors for variables so perhaps it needs to include enum names as well?

**Comment by aardappel** [comment]
Yup it should. Lobster doesn't use a central table for all idents, instead it has separate ones per language features.. will look into how to best solve this.
